ethstats/ethstats.go

```diff
@@ -92,21 +99,28 @@ func New(url string, ethServ *eth.Ethereum, lesServ *les.LightEthereum) (*Servic
 		return nil, fmt.Errorf("invalid netstats url: \"%s\", should be nodename:secret@host:port", url)
 	}
 	// Assemble and return the stats service
-	var engine consensus.Engine
+	var (
+		engine consensus.Engine
+		id     string
+	)
 	if ethServ != nil {
+		etherBase, _ := ethServ.Etherbase()
```

Comment:
worth doing `id =` outside the `if`, checking err here and only setting it to the address if `err == nil` ?